lib/Dialect/FIRRTL/Transforms/Lint.cpp

```diff
@@ -0,0 +1,63 @@
+//===- Lint.cpp -------------------------------------------------*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "PassDetails.h"
+#include "mlir/IR/Threading.h"
+#include "llvm/ADT/APSInt.h"
+
+using namespace mlir;
+using namespace circt;
+using namespace firrtl;
+
+namespace {
+struct LintPass : public LintBase<LintPass> {
+  void runOnOperation() override {
+    auto fModule = getOperation();
+    for (auto &op : fModule.getOps())
+      if (checkAssert(&op).failed())
+        return signalPassFailure();
+
```

Comment:
Can you change to inspect nested regions using `walk`? Probably it's necessary to skip WhenOp but it will allow the pass to be ran in the early pipeline

Reply:
If we want to support whens, we need to ensure all the enclosing when blocks have constant true as the condition.
The enable to the assert must be true, to detect static asserts. This pass avoids the analysis, by avoiding nested regions. What do you think ?

Reply:
I missed Group ops, that makes sense, updated to walk over nested regions.